### PR TITLE
Role Home Page

### PR DESCRIPTION
**Pull request by OHaiDanny**
This PR comes with

- Hugo templating for roles landing page
- Netlify CMS editor configurations to generate correct content for role page to populate
- Mobile Responsiveness

image

### CODE REVIEW

File: layouts/partials/cards/job_card/job_card.html [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  (.siteVar.GetPage $levelingGuide)
+  (.siteVar.GetPage $basicGuide)
+  (.siteVar.GetPage $advancedGuide)
+  (.siteVar.GetPage $skillsOverview)
+  (.siteVar.GetPage $openers)
+  (.siteVar.GetPage $faq)
+  (.siteVar.GetPage $bis)
+  (.siteVar.GetPage $jobChanges)
+  (.siteVar.GetPage $statPriority)
+}}
```
Comment (OHaiDanny) [inline layouts/partials/cards/job_card/job_card.html:24]: I tried to run a when loop to do these GetPage, but for some reason it returns an error. Sorry for the duplicates

File: tailwind.config.js [hunk 2]
```diff
@@ -41,9 +41,23 @@ module.exports = {
...
       height: {
+        '68': '17rem',
+        '84': '21rem',
+        '88': '22rem',
         '100': '25rem', 
+        '104': '26rem',
+        '108': '27rem',
         '112': '28rem',
+        '116': '29rem',
         '120': '30rem',
```
Comment (OHaiDanny) [inline tailwind.config.js:61]: There is something weird going on with Tailwind. These height values should be baked in but everytime when I try to use it, it fails to recognize. I had to put these in for it to work...

Did something happen recently that changed some tailwind configs? Maybe our @apply is being overused and it is affecting our tailwind classes?

File: exampleSite/static/admin/config.yml [hunk 3]
```diff
@@ -184,9 +186,14 @@ collections:
           widget: "select"
           options: ["tanks"]
           default: ["tanks"]
+        - {label: "Layout", name: "layout", widget: "select", options: ["role_home"], default: ["role_home"]}
+        - {label: "Background Header Image", name: "bg_header_image", widget: "image", choose_url: false}
+        - {label: "Role Hero Image", name: "role_hero_image", widget: "image", choose_url: false}
       - label: "Healers"
```
Comment (OHaiDanny) [inline exampleSite/static/admin/config.yml:192]: In each of the roles I've added

- Media Folder path
- Public Folder path
- Dropdown to select the role_home layout. It is the only one available to guarantee.
- Background Header Image
- Hero Image

The role icon with the border should already be in balance-static

File: layouts/partials/cards/job_card/job_card.html [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+          {{ .Content }}
+        {{end}}
+      </div>
+      <div>
+        <h4 class="uppercase tracking-widest border-t border-black my-4 pt-4">Guides</h4>
+        <div class="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3">
+          {{ range $pages }}
+            {{ if .Path }}
+              <div>
+                <a href="{{ .Path }}" class="text-link-orange">{{ .Name }}</a>
```
Comment (ductapeofdoom) [inline layouts/partials/cards/job_card/job_card.html:45]: `.Permalink` is probably better to use here and should always exist as well.

**Review by ductapeofdoom — CHANGES_REQUESTED** [review]
Looks good overall to me. I think there are just a few small changes that may make it bit more maintainable. Awesome job!

File: layouts/jobs/role_home.html [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+
+{{ $params := .Page.Params }}
+{{ $roleData := (index .Site.Data.roles $params.role )}}
+
+{{ partial "role/header.html" . }}
+<div class="responsive-container mt-8 lg:mt-14">
+    <h1 class="text-4xl uppercase font-bold tracking-wide">{{ $roleData.nameSingular }} Jobs</h1>
+    <div class="grid grid-cols-1 md:grid-cols-2 md:gap-4 items-start">
+      {{ range $roleData.jobs }}
+          {{ partial "cards/job_card/job_card.html" (dict "jobData" . "siteVar" $.Site "role" $roleData.name) }}
```
[the remark this answers is not in the document]
Comment (OHaiDanny) [reply layouts/jobs/role_home.html:11]: it is global until you go into partials. Once you are in partials the `$.Site` will return null. There are articles about that in which that is how I discovered this.

Unless there is something that allows you to use `$.Site`, you have to pass it down

File: layouts/partials/cards/job_card/job_card.html [hunk 6]
```diff
@@ -0,0 +1,56 @@
+
+{{ $path := printf "%s%s%s%s%s" "/content/jobs" "/" (lower .role) "/" (urlize (lower .jobData.name)) }}
```
[the remark this answers is not in the document]
Comment (OHaiDanny) [reply layouts/partials/cards/job_card/job_card.html:2]: Ohh interesting. I'll try that out

File: layouts/partials/cards/job_card/job_card.html (hunk 1 above)
Comment (OHaiDanny) [reply layouts/partials/cards/job_card/job_card.html:24]: Ohhhh you have to use `( )` like that. Interesting

File: layouts/jobs/role_home.html (hunk 5 above)
[the remark this answers is not in the document]
Comment (ductapeofdoom) [reply layouts/jobs/role_home.html:11]: Ahh ok I didn't realize that. Good to know.

File: layouts/partials/cards/job_card/job_card.html (hunk 1 above)
Comment (OHaiDanny) [reply layouts/partials/cards/job_card/job_card.html:24]: Ok I found a nice way to do it

File: layouts/partials/cards/job_card/job_card.html (hunk 4 above)
Comment (OHaiDanny) [reply layouts/partials/cards/job_card/job_card.html]: Resolved

File: layouts/partials/cards/job_card/job_card.html (hunk 6 above)
[the remark this answers is not in the document]
Comment (OHaiDanny) [reply layouts/partials/cards/job_card/job_card.html]: Resolved

File: tailwind.config.js (hunk 2 above)
Comment (nonowazu) [reply tailwind.config.js:61]: We did just recently enabled purgecss in [another PR](https://github.com/The-Balance-FFXIV/glam/pull/194) — do you think this is impacting that?

Comment (OHaiDanny) [reply tailwind.config.js:61]: @nonowazu Ahhh I think that may be it. Reading the purge css documentation there is a way for us to create a safelist. Maybe I'll look into refactoring it later

**Pull request merged**